src/common/api.js

```diff
@@ -24,28 +24,33 @@ export function callApi(path) {
     });
 }
 
-export function* fetchApi(requestTypes, path, action) {
+export function* fetchApi(requestTypes, path, method, action) {
   try {
     path = format(path, action)
-    const data = yield call(callApi, path);
+    const options = getFetchOptions(method, action);
+    const data = yield call(callApi, path, options);
     yield put({type: requestTypes.SUCCESS, data});
   } catch (error) {
     yield put({type: requestTypes.FAILURE, error});
   }
 }
 
-export function* watchApi(requestTypes, path) {
-  yield* takeLatest(requestTypes.REQUEST, fetchApi, requestTypes, path)
+export function* watchApi(requestTypes, path, method = 'get') {
+  yield* takeLatest(requestTypes.REQUEST, fetchApi, requestTypes, path, method)
 }
 
 export function reduceApi(state, action, requestTypes, onSuccess) {
```

Comment:
define `let fetchState = {};` for each case block( for a little performance gain).